scilpy/image/labels.py

```diff
@@ -91,6 +95,12 @@ def get_labels_from_mask(mask_data, labels=None, background_label=0):
     """
     # Get the number of structures and assign labels to each blob
     label_map, nb_structures = ndi.label(mask_data)
+    if min_voxel_count:
+        for label in range(1, nb_structures + 1):
+            if np.count_nonzero(label_map == label) < min_voxel_count:
+                label_map[label_map == label] = 0
+        label_map, nb_structures = ndi.label(mask_data)
```

Comment:
I think this should not be there.

Reply:
It relabels lesions after discarding the small one, this is to avoid lesions in the thousands despite having maybe a few hundred 'valid lesions'.

This is inside the `if` so it is done only if there is a threshold, I could also add a True/False in the for loop to having recomputing if there was no 'invalid lesions'

Reply:
Which is faster / clearer? Your version, or:
```
if min_voxel_count:
        new_count = 0
        for label in range(1, nb_structures + 1):
            if np.count_nonzero(label_map == label) < min_voxel_count:
                label_map[label_map == label] = 0
            else:
                  new_count +=1
                  label_map[label_map == label] = new_count
```

Reply:
I would also add a :
`logging.debug("Ignored blob with x voxels")` in the loop.